src/node/rpc/frontend.h

```diff
@@ -258,20 +258,36 @@ namespace ccf
         if (!endpoint->authn_policies.empty())
         {
           std::string auth_error_reason;
+          std::vector<ccf::ODataErrorDetails> errordetails;
           for (const auto& policy : endpoint->authn_policies)
           {
             identity = policy->authenticate(tx, ctx, auth_error_reason);
             if (identity != nullptr)
             {
               break;
             }
+            else
+            {
+              // Collate error details
+              errordetails.push_back(
+                {policy->SECURITY_SCHEME_NAME,
```

Comment:
I don't think this will actually work? The type of `policy` here is `std::shared_ptr<AuthPolicy>`, so `policy->SECURITY_SCHEME_NAME` is always looking up the static member in `AuthPolicy`, not the derived classes. You'll need a `virtual std::string get_scheme_name()` or similar to correctly override this.

Quick demo here:
https://coliru.stacked-crooked.com/a/51a2c0da21e1ee7c

Reply:
Ah let me check. I did initially have a `get_scheme_name`, but thought this would work.

Reply:
Good catch, there is an explicit method to return the name now.